CalibTracker/SiPixelLorentzAngle/src/SiPixelLorentzAnglePCLHarvester.cc

```diff
@@ -71,11 +75,22 @@ void SiPixelLorentzAnglePCLHarvester::beginRun(const edm::Run& iRun, const edm::
 
   PixelTopologyMap map = PixelTopologyMap(geom, tTopo);
   hists.nlay = geom->numberOfLayers(PixelSubdetector::PixelBarrel);
-
+  hists.nModules_.resize(hists.nlay);
   for (int i = 0; i < hists.nlay; i++) {
     hists.nModules_[i] = map.getPXBModules(i + 1);
   }
 
+  std::string modulename;
+  if (!newmodulelist_.empty()) {
+    for (auto const& string : newmodulelist_) {
+      modulename = string;
+      PixelBarrelNameUpgrade bn(modulename);
```

Comment:
this is sufficient for the moment, but what about if we have to include later on any FPix module? I think there should be a branching in the code here.